model/utils.go

```diff
@@ -155,15 +155,20 @@ func NewRandomTeamName() string {
 	return teamName
 }
 
+// NewRandomString returns a random string of the given length.
+// The resulting entropy will be (5 * length) bits.
 func NewRandomString(length int) string {
-	var b bytes.Buffer
-	str := make([]byte, length+8)
-	rand.Read(str)
-	encoder := base32.NewEncoder(encoding, &b)
-	encoder.Write(str)
-	encoder.Close()
-	b.Truncate(length) // removes the '==' padding
-	return b.String()
+	data := make([]byte, 1+(length*5/8))
+	rand.Read(data)
+	return encoding.EncodeToString(data)[:length]
```

Comment:
In fact it would be great to have `NoPadding` configured directly on `encoding` to not have to set it separately here. Then we could also remove the truncation from `NewId` and rename the `encoding` variable more descriptively to `zBase32`, because with `NoPadding` it really would be a self-contained z-base-32 encoder.

But that might be a separate PR.

Reply:
The data gets truncated, on purpose, before any padding. `NewRandomString()` expects the output to be of the given `length`.   
We can't guarantee a length with base-32 unless we make sure to generate more than enough and then truncate.
The length (without padding) for an input of length N (in bytes) will be `(N*8 + 4) / 5`.  
This means we could never get lengths of `1`, `3`, `6`, `9`, `11` and so on. A practical [example](https://play.golang.org/p/0NlxIFOYg3z).

Reply:
Hmm... ? Surely 12 bytes encodes to 20 characters? https://play.golang.org/p/ZnD1tlcuTQa

Reply:
Sorry, right. Correct you are, I just misread your example. For arbitrary lengths we do need truncation, yes.